omod-2.0/src/test/java/org/openmrs/module/webservices/rest/resource/SubDetailsResource.java

```diff
@@ -63,7 +63,12 @@ public void purge(SubDetails delegate, RequestContext context) throws ResponseEx
 	public DelegatingResourceDescription getRepresentationDescription(Representation rep) {
 		return new DelegatingResourceDescription();
 	}
-	
+
+	@Override
+	public DelegatingResourceDescription getCreatableProperties() throws ResourceDoesNotSupportOperationException {
```

Comment:
Why did we add this one?

Reply:
This class has a getCREATEModel method that needs to be removed. I wanted to replace that with getcreatableproperties since the new mechanism depends on that

Reply:
I do not seem to see where you removed it in this pull request.

Reply:
True dat. Will fix that

Reply:
I tried addressing it in [this commit](https://github.com/openmrs/openmrs-module-webservices.rest/pull/638/commits/bcaf91a7c68266080580c33181899bff9a95b6bd) but it fails some tests in swaggerspecificationcreatortest so I decided to put it back in this most recent commit.

Its really because we need that method to add `SubDetailsResource` to the swagger documentation. When I remove it, the tests fail because that resource is not added

Reply:
I still somehow feel we should not add this new method just for the sake of the javadocs.

Reply:
@dkayiwa @ibacher, Are you suggesting i remove even the [SubDetailsResource](https://github.com/openmrs/openmrs-module-webservices.rest/blob/5ab2a24cc62c1b3eb81812b591420336ca8d9098/omod-2.0/src/test/java/org/openmrs/module/webservices/rest/resource/SubDetailsResource.java#L33)? because that's the only way the existing tests especially [this one](https://github.com/openmrs/openmrs-module-webservices.rest/blob/5ab2a24cc62c1b3eb81812b591420336ca8d9098/omod-2.0/src/test/java/org/openmrs/module/webservices/rest/doc/SwaggerSpecificationCreatorTest.java#L237) will pass?.

And to be frank, SubDetailsResource is just a test class that shouldn't appear in our swagger spec but apparently does so i am okay with removing it.

Reply:
Hmmm... it shouldn't appear in the production spec, but I thought it was there so the tests have something to work with. I don't think we should remove it, but I do think we should re-work the `SwaggerSpecificationCreatorTest` class so that it tests whatever mechanism we're using to generate the Swagger Specification now.

We don't need existing tests to pass if we're changing the mechanism, but we need an equivalent set of tests to pass. Does that make sense?

Also, when actually generating the spec, we should ensure that none of the `test` modules are on the classpath.